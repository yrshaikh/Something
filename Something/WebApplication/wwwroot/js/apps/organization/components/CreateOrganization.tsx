import * as React from "react";
import { Alert, Button, Form, Input } from "antd";
import { OrganizationService } from "../../services/OrganizationService";

const classNames = require("classnames");
const FormItem = Form.Item;

import "./CreateOrganization.scss";

export class CreateOrganization extends React.Component<CreateOrganization.Props, CreateOrganization.State> {
    public static displayName = "CreateOrganization";
    constructor(props: CreateOrganization.Props) {
        super(props);
        this.state = {
            hasError: false,
            organizationName: "",
        };
        this.onChangeOrgnaizationName = this.onChangeOrgnaizationName.bind(this);
        this.createOrganization = this.createOrganization.bind(this);
    }

    public render(): JSX.Element {
        return (
            <div className="ant-row">
                {this.renderForm()}
                <hr className="ant-col-md-24 CreateOrganization__hr"/>
                {this.renderError()}
            </div>
        );
    }

    private renderForm(): JSX.Element {
        return <Form
            className="ant-col-md-24 CreateOrganization"
        >
            <p>
                <b>
                    Get most out of your organization
                </b>
            </p>
            <p>Organization helps you to group users and projects. You can create mulitple organization, one user can be linked with mulitple organizations.</p>
            <b>Organization name</b>
            <FormItem>
                <Input
                    className="ant-input-lg CreateOrganization__input"
                    placeholder="ACME INC"
                    name="Email"
                    onChange={this.onChangeOrgnaizationName}
                    autoFocus
                />
            </FormItem>
            <FormItem>
                <Button
                    type="primary"
                    htmlType="submit"
                    className={this.getButtonCssClasses()}
                    onClick={this.createOrganization}
                >
                    Create Organization
                </Button>
            </FormItem>
        </Form>;
    }

    private renderError(): JSX.Element {
        return this.state.hasError ?
            <Alert
                className="ant-col-md-24"
                message="Oops! Something went wrong."
                description="There was an error creating your orgranization. Please try again in sometime."
                type="error"
            /> : null;
    }

    private getButtonCssClasses(): string {
        return classNames("login-form-button", "ant-btn--rounded",
            { "ant-btn--disabled": !(this.state.organizationName && this.state.organizationName.length >= 3) },
        );
    }

    private onChangeOrgnaizationName(e: any) {
        const name = e.target.value;
        this.setState(function (state, props) {
            return {
                organizationName: name
            }
        });
    }

    private createOrganization() {
        const that = this;
        OrganizationService.post(this.state.organizationName)
            .then((response) => response.text())
            .then((organizationId) => that.onOrganizationCreated(organizationId))
            .catch((response) => that.setState({hasError: true}));
    }

    private onOrganizationCreated(organizationId: number) {
        if (this.props.isOnboarding) {
            this.props.onCreateCallback(CreateOrganization.displayName);
        }
    }

}
