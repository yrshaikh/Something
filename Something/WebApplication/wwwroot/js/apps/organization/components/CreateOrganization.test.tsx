import "jest";
import * as React from "react";
import { shallow } from "enzyme";
import enzymeToJson from "enzyme-to-json";

import { CreateOrganization } from "./CreateOrganization";

describe("<CreateOrganization />", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<CreateOrganization isOnboarding={true} onCreateCallback={null} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it("should enable button when text is entered", () => {
        const wrapper = shallow(<CreateOrganization isOnboarding={true} onCreateCallback={null} />);
        expect(wrapper.find('.ant-btn--disabled').length).toBe(1);
        wrapper.setState({ organizationName: 'howdy' });
        expect(wrapper.find('.ant-btn--disabled').length).toBe(0);
    });
});
