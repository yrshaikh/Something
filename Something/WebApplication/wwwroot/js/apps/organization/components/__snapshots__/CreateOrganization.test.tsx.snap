// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateOrganization /> should render correctly 1`] = `
<div
  className="ant-row"
>
  <Form
    className="ant-col-md-24 CreateOrganization"
    colon={true}
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
  >
    <p>
      <b>
        Get most out of your organization
      </b>
    </p>
    <p>
      Organization helps you to group users and projects. You can create mulitple organization, one user can be linked with mulitple organizations.
    </p>
    <b>
      Organization name
    </b>
    <FormItem
      hasFeedback={false}
    >
      <Input
        autoFocus={true}
        className="ant-input-lg CreateOrganization__input"
        disabled={false}
        name="Email"
        onChange={[Function]}
        placeholder="ACME INC"
        type="text"
      />
    </FormItem>
    <FormItem
      hasFeedback={false}
    >
      <Button
        block={false}
        className="login-form-button ant-btn--rounded ant-btn--disabled"
        ghost={false}
        htmlType="submit"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Create Organization
      </Button>
    </FormItem>
  </Form>
  <hr
    className="ant-col-md-24 CreateOrganization__hr"
  />
</div>
`;
